groups/reconcile.go

```diff
@@ -311,12 +421,19 @@ func (gc *GroupsConfig) Load(rootDir string, restrictions *RestrictionsConfig) e
 	log.Printf("reading groups.yaml files recursively at %s", rootDir)
 
 	return filepath.Walk(rootDir, func(path string, info os.FileInfo, err error) error {
+		// check if the current path is accessible or not.
+		if err != nil {
+			return err
+		}
 		if filepath.Base(path) == "groups.yaml" {
 			cleanPath := strings.Trim(strings.TrimPrefix(path, rootDir), string(filepath.Separator))
 			log.Printf("groups: %s", cleanPath)
 
-			var groupsConfigAtPath GroupsConfig
-			var content []byte
+			var (
+				groupsConfigAtPath GroupsConfig
+				content            []byte
+				err                error
+			)
```

Comment:
I am not used to seeing this style of var declaration outside of globals but maybe that's just me